src/number_theory/padics/padic_integers.lean

```diff
@@ -579,6 +578,27 @@ lemma ideal_eq_span_pow_p {s : ideal ℤ_[p]} (hs : s ≠ ⊥) :
   ∃ n : ℕ, s = ideal.span {p ^ n} :=
 discrete_valuation_ring.ideal_eq_span_pow_irreducible hs irreducible_p
 
+open cau_seq
+
+instance : is_adic_complete (maximal_ideal ℤ_[p]) ℤ_[p] :=
+{ prec' := λ x hx,
+  begin
+    simp only [← ideal.one_eq_top, smul_eq_mul, mul_one, smodeq.sub_mem, maximal_ideal_eq_span_p,
+      ideal.span_singleton_pow, ← norm_le_pow_iff_mem_span_pow] at hx ⊢,
+    let x' : cau_seq ℤ_[p] norm := ⟨x, _⟩, swap,
```

Comment:
You could remove this `swap` and, erm, swap the order of the proofs :-)

Reply:
Sure, but I think it's natural to do it in this order. I claim that `x` is a Cauchy sequence, so better explain why that's true.

Reply:
I agree it's more natural if you're reading the proofs :-)